app/javascript/components/App.js

```diff
@@ -131,6 +136,20 @@ export default class App extends React.Component{
             }}
           />
 
+          <Route path="/lend/:id"
+            render={(props) => {
+              let id = props.match.params.id
+              let book = this.state.books.find(book => book.id === parseInt(id))
```

Comment:
Since you are doing the same thing on line 93, you could extract out a helper method?